server/src/main/kotlin/fi/oph/kitu/yki/YkiService.kt

```diff
@@ -21,37 +24,54 @@ class YkiService(
         lastSeen: LocalDate? = null,
         dryRun: Boolean? = null,
     ) {
-        val event = logger.atInfo()
-        event.addKeyValue("dryRun", dryRun)
-        event.addKeyValue("lastSeen", lastSeen)
+        logger
+            .atInfo()
+            .addKeyValue("dryRun", dryRun)
+            .addKeyValue("lastSeen", lastSeen)
+            .log("start yki import")
 
         try {
-            val suoritukset =
+            val response =
                 ykiRestClient
                     .get()
                     .uri("suoritukset")
                     .retrieve()
-                    .body(String::class.java)
-                    ?.asCsv<YkiSuoritus>() ?: throw RestClientException("Response body is empty")
+                    .toEntity<String>()
+
+            logger
+                .atInfo()
+                .addResponse(response)
+                .addKeyValue("external-system", "YKI")
+                .log("YKI suoritukset response")
```

Comment:
Onko siitä hyötyä, että nämä ovat omia lokirivejään? Jos `event` luotaisiin try-catch-finally:n alussa, voitaisiin periaatteessa liittää kaikki lokitieto siihen yhteen eventtiin ja kutsua sille `.log()` vasta `finally`:ssa. Näin meillä olisi yhtä import-operaatiota kohden yksi lokirivi (tai kaksi, mikäli "aloitetaan" ja "valmis" -rivit halutaan erikseen), joka sisältäisi kaiken operaatioon liittyvän tiedon.

Myös `catch`-lohko voisi käyttää samaa `event`-instanssia, jolloin virhetilanteessa lokiriville tulisi silti kaikki operaation aikana kertynyt lokitieto, aina virhehetkeen asti.

Reply:
Mä en tiedä mitä etuja yksittäisistä on vs useista. Mä ajattelin, että lokitan
 1. `tapahtuma alkoi`
 2. `vastaus ulkoisesta rajapinnasta`
 3. `tapahtuma loppui` tai `tapahtumassa virhe`

Onko tähän olemassa jotain best practiseja? Tai pitäiskö meidän sopia jotain käytäntöjä?

Reply:
Kysyin vielä chatGPT:ltä ja se vastas


> Single Log Event with Attributes
>
>   Pros:
>        Reduces the number of log entries, simplifying log management.
>        Easier to track the entire operation in a single entry.
>
>    Cons:
>        Lack of granularity makes it harder to pinpoint where an issue occurred.
>        May become verbose if you include too much detail in a single log entry.
>
>Multiple Log Events
>
>    Pros:
 >       Provides detailed insights into each step of the process, making it easier to identify failures.
>        Facilitates monitoring and alerting based on specific events (e.g., HTTP responses).
>
>    Cons:
>        Can generate a large volume of logs, which may lead to log management challenges.
>        Potential performance overhead due to increased logging activity.